example/extensions/lib_subgraph/subgraph_lib.cc

```diff
@@ -0,0 +1,278 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2019 by Contributors
+ * \file subgraph_lib.cc
+ * \brief subgraph operator implementation library file
+ */
+
+#include <math.h>
+#include <iostream>
+#include <algorithm>
+#include "lib_api.h"
+
+MXReturnValue parseAttrs(std::map<std::string, std::string> attrs,
```

Comment:
If we do not care about the order of keys within the map, how about replacing std::map with unordered_map?

Reply:
i guess we could, can you make this suggestion in the RFC for custom op enhancements and mention how would that help here, #17006